sig/lrama/lexer/token.rbs

```diff
@@ -0,0 +1,47 @@
+module Lrama
+  class Lexer
+    class Token
+
+      attr_accessor line: Integer
+      attr_accessor column: Integer
+      attr_accessor referred: bool
+      attr_accessor references: Array[[Symbol, (String|Integer), Token?, Integer, Integer]]
+
+      def to_s: () -> String
+      def referred_by?: (String string) -> bool
+      def ==: (Token other) -> bool
```

Comment:
I'm only worried about the argument type of the `==` method.

Reply:
The code can handle other type objects however I expect `other` is Token. Therefore let's try `Token` for the argument type.